### ISSUE TITLE
firefox NS_ERROR_FAILURE

### ISSUE THREAD

**Issue by mafar** [ismailhabib/custom-protocol-detection#24]
Consider change below. This is because if protocol is registered but app is not there then firefox throws another error e.name==NS_ERROR_FAILURE

```
if (e.name == "NS_ERROR_UNKNOWN_PROTOCOL") {
            failCb();
}
```
to
```
if (e ) {
            failCb();
}
```

**Comment by edwardstumpf** [comment]
It would be better to only trigger the failure callback for exceptions related to protocols.  The successCb function may also throw an exception, so you wouldn't want to call failCb in that case.

```
} catch (e) {
  // Catch errors for either unknown protocols or registered protocols where the associated application isn't installed
  if (e.name == "NS_ERROR_UNKNOWN_PROTOCOL" || e.name == "NS_ERROR_FAILURE") {
    failCb();
  }
}
```

**Comment by mafar** [comment]
@edwardstumpf 
Yes may be but NS_ERROR_FAILURE indicates protocol might be reigstered but app could not be launched for some reason ( may be exe was moved or does not exist at that path)

See another implementation here.
https://gist.github.com/aaronk6/d801d750f14ac31845e8

They put it in unknown callback for this reason

**Comment by edwardstumpf** [comment]
I agree that the failure callback should be triggered if the exception throw is NS_ERROR_FAILURE.  As you said, this indicates that the application associated with the protocol isn't at the expected location.  This could mean that there's an issue with the custom protocol (the wrong path for the application was used) or the application doesn't exist at that location (uninstalled, etc.)

I don't agree that all exception name checking should be removed.  The success callback could also throw an exception; this exception would have nothing to do with protocol checking, so the library shouldn't catch that exception.
